openwisp_utils/bots/ci_failure/bot.py

```
@@ -0,0 +1,464 @@
+            if not self.github_token:
+                missing.append("GITHUB_TOKEN")
+            if not self.repository_name:
+                missing.append("REPOSITORY")
+            print(f"Warning: Missing environment variables: {', '.join(missing)}")
+
+        if self.gemini_api_key:
+            try:
+                genai.configure(api_key=self.gemini_api_key)
+                self.model_name = os.environ.get("GEMINI_MODEL", "gemini-2.5-flash")
+                self.model = genai.GenerativeModel(self.model_name)
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)